influxql/point_test.go

```diff
@@ -6,8 +6,155 @@ import (
 	"testing"
 
 	"github.com/influxdb/influxdb/influxql"
+	"github.com/influxdb/influxdb/pkg/deep"
 )
 
+func TestPoint_Clone_Float(t *testing.T) {
+	p := &influxql.FloatPoint{
+		Name:  "cpu",
+		Tags:  ParseTags("host=server01"),
+		Time:  5,
+		Value: 2,
+		Aux:   []interface{}{float64(45)},
+	}
+	c := p.Clone()
+	if p == c {
+		t.Errorf("clone has the same address as the original: %v == %v", p, c)
+	}
+	if p.Name != c.Name {
+		t.Errorf("mismatched name: %v != %v", p.Name, c.Name)
+	}
+	if !deep.Equal(p.Tags, c.Tags) {
+		t.Errorf("mismatched tags: %v != %v", p.Tags.ID(), c.Tags.ID())
+	}
+	if p.Time != c.Time {
+		t.Errorf("mismatched time: %v != %v", p.Time, c.Time)
+	}
+	if p.Value != c.Value {
+		t.Errorf("mismatched value: %v != %v", p.Value, c.Value)
+	}
+	if &p.Aux[0] == &c.Aux[0] {
+		t.Errorf("aux values share the same address: %v == %v", p.Aux, c.Aux)
+	} else if !deep.Equal(p.Aux, c.Aux) {
+		t.Errorf("mismatched aux fields: %v != %v", p.Aux, c.Aux)
+	}
+}
+
+func TestPoint_Clone_Integer(t *testing.T) {
+	p := &influxql.IntegerPoint{
+		Name:  "cpu",
+		Tags:  ParseTags("host=server01"),
+		Time:  5,
+		Value: 2,
+		Aux:   []interface{}{float64(45)},
+	}
+	c := p.Clone()
+	if p == c {
+		t.Errorf("clone has the same address as the original: %v == %v", p, c)
+	}
+	if p.Name != c.Name {
+		t.Errorf("mismatched name: %v != %v", p.Name, c.Name)
+	}
+	if !deep.Equal(p.Tags, c.Tags) {
+		t.Errorf("mismatched tags: %v != %v", p.Tags.ID(), c.Tags.ID())
+	}
+	if p.Time != c.Time {
+		t.Errorf("mismatched time: %v != %v", p.Time, c.Time)
+	}
+	if p.Value != c.Value {
+		t.Errorf("mismatched value: %v != %v", p.Value, c.Value)
+	}
+	if &p.Aux[0] == &c.Aux[0] {
+		t.Errorf("aux values share the same address: %v == %v", p.Aux, c.Aux)
+	} else if !deep.Equal(p.Aux, c.Aux) {
+		t.Errorf("mismatched aux fields: %v != %v", p.Aux, c.Aux)
+	}
+}
+
+func TestPoint_Clone_String(t *testing.T) {
+	p := &influxql.StringPoint{
+		Name:  "cpu",
+		Tags:  ParseTags("host=server01"),
+		Time:  5,
+		Value: "clone",
+		Aux:   []interface{}{float64(45)},
+	}
+	c := p.Clone()
+	if p == c {
+		t.Errorf("clone has the same address as the original: %v == %v", p, c)
+	}
+	if p.Name != c.Name {
+		t.Errorf("mismatched name: %v != %v", p.Name, c.Name)
+	}
+	if !deep.Equal(p.Tags, c.Tags) {
+		t.Errorf("mismatched tags: %v != %v", p.Tags.ID(), c.Tags.ID())
+	}
+	if p.Time != c.Time {
+		t.Errorf("mismatched time: %v != %v", p.Time, c.Time)
+	}
+	if p.Value != c.Value {
+		t.Errorf("mismatched value: %v != %v", p.Value, c.Value)
+	}
+	if &p.Aux[0] == &c.Aux[0] {
+		t.Errorf("aux values share the same address: %v == %v", p.Aux, c.Aux)
+	} else if !deep.Equal(p.Aux, c.Aux) {
+		t.Errorf("mismatched aux fields: %v != %v", p.Aux, c.Aux)
+	}
+}
```

Comment:
Why not do a single `reflect.DeepEqual()` check and then `spew` the whole point if it doesn't match?